src/thor/timedistancematrix.cc

```diff
@@ -541,10 +583,14 @@ bool TimeDistanceMatrix::UpdateDestinations(
 }
 
 // Form the time, distance matrix from the destinations list
-std::vector<TimeDistance> TimeDistanceMatrix::FormTimeDistanceMatrix() {
+std::vector<TimeDistance> TimeDistanceMatrix::FormTimeDistanceMatrix(GraphReader& reader,
+                                                                     const std::string origin_dt,
+                                                                     const uint64_t origin_tz,
+                                                                     const GraphId& pred_id) {
   std::vector<TimeDistance> td;
   for (auto& dest : destinations_) {
-    td.emplace_back(dest.best_cost.secs, dest.distance);
+    auto date_time = GetDateTime(origin_dt, origin_tz, pred_id, reader, dest.best_cost.secs);
+    td.emplace_back(dest.best_cost.secs, dest.distance, date_time);
```

Comment:
here I try to add the final `date_time` to the results here. few things worth noting:
1. I have historical traffic on some dataset I generated ages ago, but somehow all node_info's along the expansion have timezone_index of 0 (i.e. invalid), no idea how that can be.. the origin location has 297 (correct)
2. we only output `date_time` in the response if there actually way time tracking going on
3. the results are NOT consistent right now (i.e. `date_time` field is off compared to request and `time` field, e.g. this request has been done with `2022-10-28T09:00` and outputs:
```
"sources_to_targets": [
		[
			{
				"date_time": "2022-10-28T08:59",
				"distance": 0.000,
				"time": 0,
				"to_index": 0,
				"from_index": 0
			},
			{
				"date_time": "2022-10-28T09:01",
				"distance": 0.729,
				"time": 126,
				"to_index": 1,
				"from_index": 0
			},
			{
				"date_time": "2022-10-28T09:06",
				"distance": 2.521,
				"time": 349,
				"to_index": 2,
				"from_index": 0
			}
		],
		[
			{
				"date_time": "2022-10-28T09:01",
				"distance": 0.720,
				"time": 151,
				"to_index": 0,
				"from_index": 1
			},
			{
				"date_time": "2022-10-28T08:59",
				"distance": 0.000,
				"time": 0,
				"to_index": 1,
				"from_index": 1
			},
			{
				"date_time": "2022-10-28T09:06",
				"distance": 2.678,
				"time": 391,
				"to_index": 2,
				"from_index": 1
			}
		],
		[
			{
				"date_time": "2022-10-28T09:04",
				"distance": 2.579,
				"time": 319,
				"to_index": 0,
				"from_index": 2
			},
			{
				"date_time": "2022-10-28T09:06",
				"distance": 2.670,
				"time": 392,
				"to_index": 1,
				"from_index": 2
			},
			{
				"date_time": "2022-10-28T08:59",
				"distance": 0.000,
				"time": 0,
				"to_index": 2,
				"from_index": 2
			}
		]
	]
```